base/show.jl

```diff
@@ -2639,6 +2639,9 @@ function array_summary(io::IO, a, inds)
     print(io, " with indices ", inds2string(inds))
 end
 
+## `summary` for Function
+summary(::Function) = "Function"
```

Comment:
I don't think
```julia
julia> summary(sum)
"Function"
```
is better than the current
```julia
julia> summary(sum)
"typeof(sum)"
```
Perhaps something like this?
```suggestion
summary(io::IO, f::Function) = show(io, MIME"text/plain"(), f)
```
This would give
```julia
julia> summary(sum)
"sum (generic function with 14 methods)"
```